### PR TITLE
docs: add workflow to lock issue conversations

### LINKED ISSUE #966: [Feature] Workflow to lock conversations
### Description

- This workflow will lock conversations. Its purpose is to lock newly created issues to prevent further modifications or comments, except in specific cases.
- Unnecessary extension of comments will be avoided.
- Comments will be unlocked when a particular label is added.
- Collaborators will have the freedom to comment.

Now, this method/workflow may be relevant based on various criteria. If you need more details or if this information is applicable to your repository, please let me know. I have also provided an overview image for reference.

One major downside is that the author will not be able to edit the issue comment. This means that if something is forgotten or needs to be added later, it cannot be modified, which can be a significant drawback.

### Screenshots

image

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by Anmol-Baranwal**
## Fixes Issue

Resolved #966 

## Changes proposed

- Reference: https://github.com/dessant/repo-lockdown

## Code

```yml
name: 'Issue Lockdown'

on:
  issues:
    types: opened

permissions:
  issues: write

jobs:
  action:
    runs-on: ubuntu-latest
    steps:
      - uses: dessant/repo-lockdown@v3 # Reference: https://github.com/dessant/repo-lockdown
        with:
          close-issue: false
          process-only: 'issues'
          issue-labels: '🔒 locked'
          exclude-issue-labels: '🚀 ready'
          skip-closed-issue-comment: true
          issue-comment: >
            To reduce notifications, issues are locked. Your issue will be unlocked when we will add label as `🚀 ready`.
```

## Note to reviewers

- The bot will add the `🔒 locked` label before locking the conversation, and you can unlock the conversation by adding the `🚀 ready` label.
- If you prefer a different label, please let me know, and I will update it accordingly.

@rupali-codes 
Please review the workflow.

### CODE REVIEW

File: .github/workflows/lock-issue.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: dessant/repo-lockdown@v3 # Reference: https://github.com/dessant/repo-lockdown
+        with:
+          close-issue: false
+          process-only: 'issues'
+          issue-labels: '🔒 locked'
+          exclude-issue-labels: '🚀 ready'
+          skip-closed-issue-comment: true
+          issue-comment: >
+            To reduce notifications, issues are locked. Your issue will be unlocked when we will add label as `🚀 ready`.
```
Comment (CBID2) [inline .github/workflows/lock-issue.yml:22]: [Proofreading] 
```suggestion
            To reduce notifications, issues are locked. Your issue will be unlocked when we add the label, `🚀 ready`.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Looks pretty good @Anmol-Baranwal! :) I just gave a suggestion on how to improve the issue comment.

**Comment by Anmol-Baranwal** [comment]
Hi @ujjawaltyagii 
Since I am planning to add another workflow, I would like to provide it to you first, considering that I have already submitted a pull request for this issue. I kindly request you to let me know if you plan to work on implementing this feature, as it will undoubtedly be beneficial for the project.

This workflow will be an automated PR labeling system that will greatly simplify the pull request management process. By analyzing the titles of PRs, it will automatically assign appropriate labels based on predefined criteria such as status, goals, and aspects. It will use `regexs`, `events`, `targets`. Since I haven't come across this workflow before, it will be a significant contribution, and I encourage you to discuss it further with @CBID2 and @rupali-codes.

To implement this workflow, you will need to use two separate workflows: one for the main code and another for defining the conditions or values of different criteria. By dividing the workflow into these two parts, it allows for better organization and maintainability of the codebase.

**Comment by CBID2** [comment]
> Hi @ujjawaltyagii Since I am planning to add another workflow, I would like to provide it to you first, considering that I have already submitted a pull request for this issue. Please let me know if you plan to add this workflow, as it will undoubtedly be great for the project.
> 
> This workflow will be an automated PR labeling system that will greatly simplify the pull request management process. By analyzing the titles of PRs, it will automatically assign appropriate labels based on predefined criteria such as status, goals, and aspects. It will use `regexs`, `events`, `targets`. Since I haven't come across this workflow before, it will be a significant contribution, and I encourage you to discuss it further with @CBID2 and @rupali-codes.
> 
> To implement this workflow, you will need to use two separate workflows: one for the main code and another for defining the conditions or values of different criteria. By dividing the workflow into these two parts, it allows for better organization and maintainability of the codebase.

Hi @Anmol-Baranwal! :) For organizational purposes, I created two separate issues for the other workflows that you just suggested.

**Review by rupali-codes — APPROVED** [review]

**Review by CBID2 — APPROVED** [review]
LGTM!

**Pull request merged**